src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -4587,6 +4587,30 @@ address generate_lookup_secondary_supers_table_stub(u1 super_klass_index) {
     return start;
   }
 
+  // load Method* target of MethodHandle
+  // R3_ARG1 = jobject receiver
+  // R19_method = result Method*
+  address generate_upcall_stub_load_target() {
+
+    StubCodeMark mark(this, "StubRoutines", "upcall_stub_load_target");
+    address start = __ pc();
+
+    __ resolve_global_jobject(R3_ARG1, R22_tmp2, R23_tmp3, MacroAssembler::PRESERVATION_FRAME_LR_GP_FP_REGS); // kills R31
```

Comment:
The comment "// kills R31" is not true. Please remove it. Can you also improve the indentation in the succeeding lines, please? Otherwise, the PPC64 code looks good and the test/jdk/java/foreign tests are passing (also with ZGC).